### ISSUE TITLE
 How to set the range of light color temperature and physical properties

### ISSUE THREAD

**Issue by cheng5230** [ambiot/ambd_matter#126]
How to Set Lamp Color Temperature Range of Physical Properties ColorTempPhysicalMaxMireds
ColorTempPhysicalMinMireds  Adapt to the maximum and minimum values ​​of color temperature under different device apps
![image](https://github.com/ambiot/ambd_matter/assets/38391461/0f576f17-372f-4619-96ff-3125027891e7)

thanks。

**Comment by xshuqun** [comment]
Do you mean to set values for your lamp according to ColorControl attribute ColorTempPhysicalMaxMireds and ColorTempPhysicalMinMireds?

Thank you.

**Comment by cheng5230** [comment]
> Do you mean to set values for your lamp according to ColorControl attribute ColorTempPhysicalMaxMireds and ColorTempPhysicalMinMireds?
> 
> Thank you.

yes ；Set the range of ColorTempPhysicalMaxMireds and ColorTempPhysical MinMireds in the ColorControl property to limit the distribution value of the app's color temperature。
 Thanks

**Comment by xshuqun** [comment]
You can adjust the range in the zap file and during runtime, you can use the following to Set and Get the attribute, if needed.

```
getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMaxMireds::Set(1, SetMaxTemp);
VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);

getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMinMireds::Set(1, SetMinTemp);
VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);

getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMaxMireds::Get(1, &GetMaxTemp);
VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);

getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMinMireds::Get(1, &GetMinTemp);
VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);
```
Thank you

**Comment by cheng5230** [comment]
> You can adjust the range in the zap file and during runtime, you can use the following to Set and Get the attribute, if needed.
> 
> ```
> getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMaxMireds::Set(1, SetMaxTemp);
> VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);
> 
> getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMinMireds::Set(1, SetMinTemp);
> VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);
> 
> getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMaxMireds::Get(1, &GetMaxTemp);
> VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);
> 
> getstatus = Clusters::ColorControl::Attributes::ColorTempPhysicalMinMireds::Get(1, &GetMinTemp);
> VerifyOrExit(getstatus == Status::Success, err = CHIP_ERROR_INTERNAL);
> ```
> 
> Thank you
log err 
chip[ZCL] 0xe1412a6 ep 1 clus 0x0000_0300 attr 0x0000_400C not supported
chip[ZCL] 0xe1412a6 ep 1 clus 0x0000_0300 attr 0x0000_400B not supported
![微信截图_20240624114940](https://github.com/ambiot/ambd_matter/assets/38391461/5e372212-eb2b-4120-8a85-bc4856dec65a)
thanks

**Comment by xshuqun** [comment]
Which endpoint have you enabled ColorControl? 
The code Get/Set(1, SetMaxTemp), 1 is the endpoint, and  SetMaxTemp is the value, you can change according to your application.

Thank you

**Comment by cheng5230** [comment]
> Which endpoint have you enabled ColorControl? The code Get/Set(1, SetMaxTemp), 1 is the endpoint, and SetMaxTemp is the value, you can change according to your application.
> 
> Thank you
endpoint 1 enabled ColorControl
![image](https://github.com/ambiot/ambd_matter/assets/38391461/9b180890-2d43-48be-8e09-1be21eb8ea2c)
thanks
